doc/source/dev_guide/contributing.rst

```diff
@@ -0,0 +1,201 @@
+to do this is to first follow the
+:doc:`conda installation instructions </installation>`
+by creating a new environment with the stable version of SIFT installed.
+
+Next we will need to uninstall the conda package of SIFT:
+
+.. code-block:: bash
+
+   conda uninstall --force uwsift
+
+This will force `uwsift` to be uninstalled without uninstalling its
+dependencies. We can then install the unstable version of SIFT from its source
+code.
+
+From GitHub
+^^^^^^^^^^^
+
+If you don't plan on making modifications to the SIFT source code then
+we can run the following command and run the `uwsift` package as usual (see
+above):
+
+.. code-block:: bash
+
+   pip install git+https://github.com/ssec/sift.git
+
+From Source
+^^^^^^^^^^^
+
+If you do plan on making changes to the source code and running SIFT to see
+the changes, you first need to clone the git repository:
+
+.. code-block:: bash
+
+   git clone git@github.com:ssec/sift.git
+
+Note the above command uses "SSH" access to GitHub which requires
+`setting up SSH keys <https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh>`_
+on your GitHub account. You can alternatively use the
+`https://github.com/ssec/sift.git` URL.
+
+Now we can install SIFT from the source code:
+
+.. code-block:: bash
+
+   cd sift
+   pip install -e . --no-deps
+
+Any changes made to the source code from here on out will take effect
+immediately.
+
+Building the Documentation
+--------------------------
+
+SIFT uses the sphinx documentation tool to generate its
+`documentation website <https://sift.readthedocs.io/en/latest/>`_.
+The website is automatically generated from the contents of the primary
+branch on GitHub. If you'd like to make changes to the documentation you can
+build the website locally to test your changes.
+In addition to the above Developer Installation process, you'll need to run
+the following commands to install sphinx-specific dependencies:
+
+.. code-block:: bash
+
+    conda install -c conda-forge sphinx sphinx_rtd_theme
+    pip install blockdiag sphinxcontrib-seqdiag sphinxcontrib-blockdiag
+
+You can then generate the documentation by running:
+
+.. code-block:: bash
+
+   cd doc
+   make html
+
+You can then open the `build/html/index.html` file in your preferred browser
+to preview the website.
+
+Additional Satpy Readers
+------------------------
+
+Starting with SIFT 1.1 Satpy is used for reading all input data files. 
+Check :doc:`the reader configuration docs </configuration/readers>` to configure which Satpy 
+readers are made available in SIFT. If you have a data source that is not yet supported in Satpy 
+(available readers are listed `here <https://satpy.readthedocs.io/en/stable/index.html#id1>`_ ), you 
+have to write a Satpy reader in order to be able to visualize it in SIFT. Head to the Satpy 
+documentation for guidelines on writing a new reader `here <https://satpy.readthedocs.io/en/stable/dev_guide/custom_reader.html>`_, and consider 
```

Comment:
On, the Sphinx build is failing now
```
Warning, treated as error:
/home/runner/work/sift/sift/doc/source/dev_guide/contributing.rst:2:Duplicate explicit target name: "here".
```
I think this should fix it? (https://github.com/sphinx-doc/sphinx/issues/3921#issuecomment-315581557)
```suggestion
(available readers are listed `here <https://satpy.readthedocs.io/en/stable/index.html#id1>`__ ), you 
have to write a Satpy reader in order to be able to visualize it in SIFT. Head to the Satpy 
documentation for guidelines on writing a new reader `here <https://satpy.readthedocs.io/en/stable/dev_guide/custom_reader.html>`__, and consider 
```

Reply:
Oh I've never seen that anonymous references idea. That's awesome. I always reworded things so I wasn't using `here`. Great!